src/stackbox/slice/mod.rs

```diff
@@ -32,8 +32,16 @@ impl<'frame, Item: 'frame> StackBox<'frame, [Item]> {
         unsafe { StackBox::assume_owns(&mut *(r as *mut Item).cast()) }
     }
 
-    /// [`Vec`]-like behavior for [`StackBox`]: pop its first item.
-    pub fn stackbox_pop(self: &'_ mut StackBox<'frame, [Item]>) -> Option<Item> {
+    /// [`VecDeque`](alloc::collections::VecDeque)-like behavior for [`StackBox`]: pop its first item.
+    ///
+    /// ```
+    /// use stackbox::Slot;
+    /// let mut slot = Slot::VACANT;
+    /// let arr = slot.stackbox([0, 1, 2]);
+    /// let mut slice = arr.into_slice();
+    /// assert_eq!(slice.pop_front(), Some(0));
+    /// ```
+    pub fn pop_front(self: &'_ mut StackBox<'frame, [Item]>) -> Option<Item> {
```

Comment:
In order to avoid conflicting with potential future APIs on `&mut [Item]`, I have been using a `.stackbox_…()` prefix for these:

```suggestion
    pub fn stackbox_pop_front(self: &'_ mut StackBox<'frame, [Item]>) -> Option<Item> {
```